client/src/utils/Modal.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { ReactElement, useContext } from 'react';
+import ReactDOM from 'react-dom';
+import Contexts from '../contexts/Contexts';
+
+export default function Modal(): ReactElement {
+  const { modalContent, setModalContent } = useContext(Contexts.ModalContext);
+
+  const onClose = () => {
+    setModalContent(null);
+  };
+
+  if (modalContent) {
+    return ReactDOM.createPortal(
+      <div className="modal-wrapper">
+        <div className="modal-backdrop"></div>
+        <Contexts.ModalContext.Provider value={{ setModalContent }}>
```

Comment:
Do we need this Modal Context Provider? Does the one we use in the main App still work here?